src/app/CommandHandler.h

```diff
@@ -587,11 +573,40 @@ class CommandHandler
         prepareParams.SetStartOrEndDataStruct(false);
 
         ScopedChange<bool> internalCallToAddResponse(mInternalCallToAddResponseData, true);
-        ReturnErrorOnFailure(PrepareInvokeResponseCommand(aResponseCommandPath, prepareParams));
+        return PrepareInvokeResponseCommand(aResponseCommandPath, prepareParams);
+    }
+
+    // TODO(#31627): It would be awesome if we could remove this template all together.
```

Comment:
Thanks for showing me that. I will take a look. I will evaluate that with the other idea I had, which is making some of the data model stuff be contained in a class and have those classes derive from base interface class that has Encode and other methods so we could eliminate the need for template

Reply:
To be a little more clear. This will be explored while working on https://github.com/project-chip/connectedhomeip/issues/31627. For now I want to take the low hanging fruit win we get with what is in this PR right now and evaluate other option in the future.